core/src/main/java/org/apache/carbondata/core/carbon/datastore/impl/btree/BTreeDataRefNodeFinder.java

```diff
@@ -240,9 +240,16 @@ private int compareIndexes(IndexKey first, IndexKey second) {
               firstNoDictionaryKeyBuffer.getShort(nonDictionaryKeyOffset + SHORT_SIZE_IN_BYTES);
           secondNodeDictionaryLength =
               secondNoDictionaryKeyBuffer.getShort(nonDictionaryKeyOffset + SHORT_SIZE_IN_BYTES);
-          compareResult = ByteUtil.UnsafeComparer.INSTANCE
-              .compareTo(first.getNoDictionaryKeys(), actualOffset, firstNoDcitionaryLength,
-                  second.getNoDictionaryKeys(), actualOffset, secondNodeDictionaryLength);
+          int minLength = Math.min(firstNoDcitionaryLength, secondNodeDictionaryLength);
```

Comment:
@kumarvishal09 please comment on this. I am not getting how actualOffset is same for both keybuffers. 
Also as I know length can be calculated by nextOffset-currentOffset. Please give the structure of data stored and how this function works